src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -682,6 +682,11 @@ internal override TypeSymbol SetUnknownNullabilityForReferenceTypes()
             return this;
         }
 
+        internal override (bool, TypeSymbol) TryMergeNullability(TypeSymbol other, VarianceKind variance)
+        {
+            return (true, this);
```

Comment:
`T` and `U` are not equivalent, even ignoring nullability, since `U` is more derived. Therefore, there is no merging of nullability across the two types.

Thanks for the test cases. I've added tests.

---
In reply to: [223831503](https://github.com/dotnet/roslyn/pull/29727#discussion_r223831503) [](ancestors = 223831503)